zerver/lib/user_topics.py

```diff
@@ -95,39 +98,58 @@ def set_topic_mutes(
         recipient_id = stream.recipient_id
         assert recipient_id is not None
 
-        add_topic_mute(
+        set_user_topic_visibility_policy(
             user_profile=user_profile,
             stream_id=stream.id,
             recipient_id=recipient_id,
             topic_name=topic_name,
-            date_muted=date_muted,
+            visibility_policy=UserTopic.MUTED,
+            last_updated=date_muted,
         )
 
 
-def add_topic_mute(
+@transaction.atomic
+def set_user_topic_visibility_policy(
```

Comment:
Having this function and another with the same name with `do_` at the front is pretty confusing. I think it's possible the best answer would be to inline this into its caller. But an alternative would be to call this something like `set_user_topic_visibility_policy_in_database`

Reply:
I think renaming it to `set_user_topic_visibility_policy_in_database` is the better option -- in favour of readability.

Let me know, if keeping it inline makes more sense. -- (Minor refactoring would be required)

Reply:
Yeah that rename sounds good.